src/Table/MakeExpandable.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { Subtract } from 'utility-types';
+
+export interface InjectedExpandableProps {
+    onClick(): void;
+    isExpand: boolean;
+}
+
+interface State {
+    isExpand: boolean;
+}
+
+export const makeExpandable = <P extends InjectedExpandableProps>(Component: React.ComponentType<P>) =>
+    class MakeExpandable extends React.PureComponent<Subtract<P, InjectedExpandableProps>, State>{
+        state: State = {
+            isExpand: false,
+        };
+
+        onClick = () => {
+            this.setState(prevState => ({ isExpand: !prevState.isExpand }));
```

Comment:
By right, should call `onClick` that pass down from parent as well.

Reply:
But now I think parent cannot pass down onClick to this comp.